service/history/execution/context.go

```diff
@@ -265,11 +267,18 @@ func NewContext(
 }
 
 func (c *contextImpl) Lock(ctx context.Context) error {
+	c.lockTime = time.Now()
```

Comment:
if there are two locks, the second one will still be able to overwrite the first one. Putting logic after mutex.Lock should work

Reply:
good point. I should move this below